### PR TITLE
Support national clouds in search

### PR DESCRIPTION
**Pull request by rakshith91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/_utils.py [hunk 1]
```diff
@@ -3,18 +3,24 @@
...
 
 def is_retryable_status_code(status_code):
     # type: (int) -> bool
     return status_code in [422, 409, 503]
 
 
-def get_authentication_policy(credential):
-    authentication_policy = policies.BearerTokenCredentialPolicy(
-        credential, *CREDENTIAL_SCOPES
+def get_authentication_policy(credential, **kwargs):
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_utils.py:16]: Maybe def get_authentication_policy(credential, is_async=False, **kwargs): ?

File: sdk/search/azure-search-documents/azure/search/documents/_utils.py [hunk 2]
```diff
@@ -3,18 +3,24 @@
...
+    audience = kwargs.get('audience', None)
+    is_async = kwargs.get('is_async', False)
+    if not audience:
+        audience = DEFAULT_AUDIENCE
+    scope = audience.rstrip('/') + '/.default'
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_utils.py:21]: CREDENTIAL_SCOPES was a set. But now scope is a single string. Do we support multiple scopes?

File: sdk/search/azure-search-documents/README.md [hunk 3]
```diff
@@ -275,6 +275,26 @@ result = search_client.upload_documents(documents=[DOCUMENT])
...
+```python
+# Create a SearchClient that will authenticate through AAD in the China national cloud.
+import os
+from azure.identity import DefaultAzureCredential
+from azure.search.documents import SearchClient
+
+index_name = "hotels"
+endpoint = os.environ["SEARCH_ENDPOINT"]
+key = os.environ["SEARCH_API_KEY"]
+credential = DefaultAzureCredential(authority=AzureAuthorityHosts.AZURE_CHINA)
```
Comment (kashifkhan) [inline sdk/search/azure-search-documents/README.md:294]: is there another import for `AzureAuthorityHosts` enum ?

File: sdk/search/azure-search-documents/azure/search/documents/_utils.py [hunk 4]
```diff
@@ -3,18 +3,24 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-from azure.core.pipeline import policies
+from azure.core.pipeline.policies import BearerTokenCredentialPolicy, AsyncBearerTokenCredentialPolicy
 
-CREDENTIAL_SCOPES = ["https://search.azure.com/.default"]
+DEFAULT_AUDIENCE = "https://search.azure.com"
```
Comment (kashifkhan) [inline sdk/search/azure-search-documents/azure/search/documents/_utils.py:8]: can the audience `kwarg` be any custom value or a set of values? If it's the latter maybe we can have an enum defined for default, US Gov, China, etc.?

File: sdk/search/azure-search-documents/azure/search/documents/_utils.py (hunk 2 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_utils.py:21]: we never supported multiple scopes at the same time :)

File: sdk/search/azure-search-documents/azure/search/documents/_utils.py (hunk 4 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_utils.py:8]: users can pass custom values here

File: sdk/search/azure-search-documents/README.md [hunk 5]
```diff
@@ -275,6 +275,26 @@ result = search_client.upload_documents(documents=[DOCUMENT])
...
+
+To authenticate in a [National Cloud](https://docs.microsoft.com/azure/active-directory/develop/authentication-national-cloud), you will need to make the following additions to your client configuration:
+
+- Set the `AuthorityHost` in the credential options or via the `AZURE_AUTHORITY_HOST` environment variable
+- Set the `audience` in `SearchClient`, `SearchIndexClient`, or `SearchIndexerClient`
+
+```python
+# Create a SearchClient that will authenticate through AAD in the China national cloud.
+import os
+from azure.identity import DefaultAzureCredential
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/README.md:288]:
```suggestion
from azure.identity import DefaultAzureCredential, AzureAuthorityHosts
```

File: sdk/search/azure-search-documents/README.md (hunk 3 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/README.md:294]: updated - good catch

**Review by xiangyan99 — COMMENTED** [review]
Does search support Azure Stack?

**Comment by rakshith91** [comment]
> Does search support Azure Stack?

no - they don't

File: sdk/search/azure-search-documents/azure/search/documents/_utils.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/_utils.py:16]: I think for get_authentication_policy method, we want to make is_async more explicit.

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**